### DISCUSSION TITLE
希望新增   批量清空队列的功能

### DISCUSSION THREAD

**Discussion by qq471754603** [rabbitmq/rabbitmq-server#6307] [Other] [answered]
希望新增一个批量勾选   批量清空队列的功能,一个一个清空点太慢了

**Comment by michaelklishin** [comment, marked as answer]
In several protocols RabbitMQ supports you can [ack multiple deliveries at once](https://www.rabbitmq.com/confirms.html#consumer-acks-multiple-parameter). And no, it is not "too slow to ack individually". What is slow is when you do not allow RabbitMQ to deliver more than one message before the previous one is acked, that is, when the [prefetch values](https://www.rabbitmq.com/confirms.html#channel-qos-prefetch) of the consumer channel is [set to 1 and not something like 256](https://blog.rabbitmq.com/posts/2014/04/finding-bottlenecks-with-rabbitmq-3-3/). Besides other things like the replication factor of the queues, whether the publisher uses confirms and [using what approach](https://www.rabbitmq.com/publishers.html#data-safety), and so on.

There is an operation to purge the queue, that is, delete all messages in Ready state in it.

[Streams](https://www.rabbitmq.com/streams.html) allow you do acknowledge N messages at once.
